packages/itwinui-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -183,8 +187,9 @@ export const DialogMain = React.forwardRef((props, ref) => {
 
   const mountRef = React.useCallback(
     (element: HTMLElement | null) => {
-      if (element) {
+      if (element && element !== mountedRef.current) {
```

Comment:
It's worth noting that this is very much a hack. We need to do a deeper investigation and properly fix the root cause.

My guess is that memoization is breaking somewhere, causing the callback ref to be executed on every re-render.

Reply:
This is good for now, but we might want to investigate deeper and fix `cloneElementWithRef` if that is the problem, because we are using it in so many places. 

Can be mentioned as an after-PR TODO.